drivers: entropy: Enable entropy on Lpc55s36

decsny opened the pull request:
Enable entropy on LPC55S36 using CSS

Commit:
drivers: Add NXP CSS entropy driver

Add entropy driver using NXP CSS
Add Kconfig and DTS binding for the NXP CSSv2

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
dts: arm: nxp: Enable css on lpc55s36

Add css to lpc55s36 device tree

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

danieldegrasse commented on west.yml:
```diff
@@ -88,7 +88,7 @@ manifest:
         - hal
     - name: hal_nxp
-      revision: 708c95825b0d5279620935a1356299fff5dfbc6e
+      revision: f3632589e8f8d15886df1d1070cba783413e4ca3
```
Can you move `west.yml` update into a separate commit, or add info to this commit message making it clear what the update here is pulling in?

danieldegrasse commented on drivers/entropy/CMakeLists.txt:
```diff
@@ -9,6 +9,7 @@ zephyr_library_sources_ifdef(CONFIG_ENTROPY_MCUX_RNG          entropy_mcux_rng.c
 zephyr_library_sources_ifdef(CONFIG_ENTROPY_MCUX_RNGA          entropy_mcux_rnga.c)
 zephyr_library_sources_ifdef(CONFIG_ENTROPY_MCUX_TRNG          entropy_mcux_trng.c)
 zephyr_library_sources_ifdef(CONFIG_ENTROPY_MCUX_CAAM          entropy_mcux_caam.c)
+zephyr_library_sources_ifdef(CONFIG_ENTROPY_MCUX_CSS          entropy_mcux_css.c)
```
nit: fix indentation

yvanderv reviewed (approved):
Looks good

mmahadevan108 commented on dts/arm/nxp/nxp_lpc55S3x_common.dtsi:
```diff
@@ -225,6 +229,12 @@
+	css: css@4003000 {
+		compatible = "nxp,lpc-css";
+		reg = <0x4003000 0x200>;
+		status="okay";
```
This should be `disabled` and `enabled` in the board that needs it.

decsny replied:
I specifically did it this way even though I know other devices do it at the board level because for all of our other entropy devices we set status okay at the soc level

mmahadevan108 commented on drivers/entropy/entropy_mcux_css.c:
```diff
@@ -0,0 +1,48 @@
+}
+
+static const struct entropy_driver_api entropy_mcux_css_api_funcs = {
+	.get_entropy = entropy_mcux_css_get_entropy};
```
Very minor, I would suggest moving the closing brace to a separate line.

danieldegrasse commented:
Could you add a commit here documenting the use of the CSS for entropy on the LPC55s36, similar to how the TRNG is [documented for the RT1064](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/mimxrt1064_evk/doc/index.rst)?

Commit:
doc: Add documentation about entropy on LPC55S36

Add to the feature table for LPC55S36 board to indicate
that the CSS can be used for entropy generation

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

dleach02 commented on boards/arm/lpcxpresso55s36/doc/index.rst:
```diff
@@ -77,6 +77,8 @@ already supported, which can also be re-used on this lpcxpresso55s36 board:
 | PWM       | on-chip    | pwm                                 |
 +-----------+------------+-------------------------------------+
+| CSS	    | on-chip	 | entropy			       |
++-----------+------------+-------------------------------------+
```
This doesn't seem to align with the above line

decsny replied:
in what sense

decsny replied:
oh, I think there are tabs instead of spaces, so it might look wrong depending how you view it. I'll fix it

stephanosio commented on dts/arm/nxp/nxp_lpc55S3x_common.dtsi:
```diff
@@ -225,6 +229,12 @@
+	css: css@4003000 {
+		compatible = "nxp,cssv2";
+		reg = <0x4003000 0x200>;
+		status="okay";
```
improper spacing
```suggestion
		status = "okay";
```

ceolin commented on drivers/entropy/entropy_mcux_css.c:
```diff
@@ -0,0 +1,55 @@
+	MCUX_CSSL_FP_FUNCTION_CALL_BEGIN(result, token, mcuxClCss_Prng_GetRandom(buffer, length));
+	if ((MCUX_CSSL_FP_FUNCTION_CALLED(mcuxClCss_Prng_GetRandom) != token) ||
+	    (result != MCUXCLCSS_STATUS_OK)) {
+		status = -ENOMSG;
```
Shouldn't it be `-EAGAIN` or maybe just `-EIO` ? I mean, we documented that we return -errno on error but `-EAGAIN` or `-EIO` looks more suitable here.

decsny replied:
sure

galak commented on dts/bindings/rng/nxp,cssv2.yaml:
```diff
@@ -0,0 +1,15 @@
+
+description: CSSv2 (Crypto Subsystem)
+
+compatible: "nxp,cssv2"
```
This name seems a little too generic.  Does `nxp,lpc-ccs-v2` make sense?

DerekSnell replied:
Hi @galak ,
I asked for this change, because this cssv2 security subsystem is used on more NXP MCUs than LPC.  We want to avoid confusion to re-use this binding on non-LPC SOCs.

Thanks

galak replied:
@DerekSnell ok, should at least be `nxp,css-v2`

DerekSnell replied:
Thanks, @galak .  We can make that change
